recipes/autodiff/meta.yaml

```
@@ -0,0 +1,47 @@
+{% set version = "0.4.0" %}
+
+package:
+  name: autodiff
+  version: {{ version }}
+
+source:
+  url: https://github.com/autodiff/autodiff/archive/v{{ version }}.tar.gz
+  sha256: 3f8246b8bc38bdfa41860c9a5e909b55793d2455edc525bdaa20b8060667b706
+
+build:
+  number: 0
+  skip: true  # [win or osx]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+
+  host:
+    - eigen
+
+  run:
+    - eigen
```

Should this be [pinned](https://docs.conda.io/projects/conda-build/en/latest/source/resources/variants.html?highlight=pin_compatible#pinning-at-the-recipe-level) to a range of `eigen` versions using something like:

```suggestion
    - {{ pin_compatible('eigen', max_pin='x.x') }}
```

This means that if this was build with `eigen==3.3.7` the runtime version must be meet the requirements `>=3.3.0` and `<3.4`. I don't know what the appropriate pinning is but it's probably one of: `x`, `x.x` or `x.x.x`.

---

Honestly, I don't know. I checked all recipes in `conda-recipes` and it seems the usage is:
~~~
  - eigen
  - eigen3
  - eigen 3.*
~~~
Maybe use `eigen 3.*`? Or just stick with `eigen` if this means the same?

---

A package called `eigen3` [doesn't exist](https://anaconda.org/conda-forge/eigen3). I've looked elsewhere and it's unusual for eigen to be a runtime dependency but I guess it is actually needed by `autodiff`?

If it is and if you expect any version of eigen 3 to work I would use:
```yaml
  host:
    - eigen
  run:
    - {{ pin_compatible('eigen', max_pin='x') }}
```

---

I missed this. I'll add:
~~~yaml
  run:
    - {{ pin_compatible('eigen', max_pin='x') }}
~~~